src/store/document-drive.ts

```diff
@@ -43,74 +48,237 @@ function debounceOperations(
     };
 }
 
+type FileNodeDocument =
+    | {
+          driveId: string;
+          documentId: string;
+          documentType: string;
+          document: Document | undefined;
+          status: 'LOADING' | 'ERROR';
+      }
+    | {
+          driveId: string;
+          documentId: string;
+          documentType: string;
+          document: Document;
+          status: 'LOADED';
+      }
+    | undefined;
+
+const documentCacheAtom = atom(new Map<string, Document>());
+
+const singletonFileNodeDocumentAtom = atom<FileNodeDocument>(undefined);
+const fileNodeDocumentAtom = atom(
+    get => get(singletonFileNodeDocumentAtom),
+    (get, set, newValue: FileNodeDocument) => {
+        const currentValue = get(singletonFileNodeDocumentAtom);
+
+        // if document will be loaded then sets
+        // the cached version while it loads
+        const documentCache = get(documentCacheAtom);
```

Comment:
Creates a cache for the fetched documents. When the user selects a document for the second time, he immediately sees the cached version and then the updated document when the new fetch request finishes.